dapps/assets/src/components/contacts/AddContact.ts

```diff
@@ -151,4 +177,64 @@ export default class AddContactComponent extends mixins(EvanComponent) {
   closePanel(): void {
     (this.$refs.addContactPanel as SwipePanelComponentClass).hide();
   }
+
+  /**
+   * Returns the permissions mapping for certain user. If nothing is shared with the user, copy from own and set all to
+   * denied.
+   *
+   * @param user: string - the user id.
+   */
+  async setupAutomaticShareData(): Promise<void> {
+    const runtime = this.getRuntime();
+    const { profile } = runtime;
+
+    await profile.loadForAccount();
+    const profileAddress = profile.profileContract.options.address;
+    // load profile data, so we can get the correct sort filters
+    const description = await profile.profileContainer.getDescription();
+    const entryKeys = Object.keys(description.dataSchema);
+    const profileData = {};
+    await Promise.all(entryKeys.map(async (key: string) => {
+      try {
+        // load account details
+        profileData[key] = await profile.getProfileProperty(key);
+      } catch (ex) {
+        runtime.logger.log(ex.message, 'error');
+      }
+    }));
+    this.shareFilters = getPermissionSortFilter(profileData);
+    // setup sharing obj
+    this.sharingObj = (await PermissionUtils.getContainerPermissionsForUser(
+      runtime,
+      {
+        containerAddress: profileAddress,
+        i18nScope: '_profile.sharing.profileContract',
+      },
+      this.idOrEmail,
+    )).permissions;
```

Comment:
if the inner await function fails it will probably crash with an unhandled error.
I think we should avoid this pattern 
```
const variable = (await asyncfunction(param1, param2)).key;
```

and use more often this:
```
const var1 = await asyncfunction(param1, param2));
const { variable } = var1;
````
it's also better to read.